### ISSUE TITLE
VSCode fails to send ctrl-c to arm-none-eabi-gdb

### ISSUE THREAD

**Issue by kanflo** [microsoft/vscode-cpptools#707]
After some debugging and reading issues such as #328 and #703 it seems to me VSCode fails to send ctrl-c to the arm-none-eabi-gdb process to break the running target.

The workaround discussed in #328 is to add a breakpoint in main(...) for the debugger to halt initially. That works for me and I can single step but if I press RUN and then PAUSE the target is not halted. Halting is still possible by telnetting to localhost 4444 and issuing the ```halt```command to OpenOCD. The target is now halted and "debuggable" in VSCode.

If I do all debugging directly in arm-none-eabi-gdb in the terminal everything works file, including breaking with ctrl-c:

```
^C
Program received signal SIGINT, Interrupt.
```

I am running Version 1.12.1 (1.12.1) (f6868fce3eeb16663840eb82123369dec6077a9b) in macOS.
Edit: I see the same issue in the exact same version in Linux.

**Comment by kanflo** [comment]
It seems the issue might reside in the MIEngine. Looking at ```CmdBreakInternal()``` in ```src/MICore/Debugger.c``` it's possible SIGINT is not sent to the arm-none-eabi-gdb process.

Too bad vscode-cpptools isn't complete open source meaning I could fix the issue myself ;)

**Comment by pieandcakes** [comment]
@kanflo MIEngine is open source and you can contribute fixes there. If you let me know when those are contributed, we can insert the fix in a future release. When you specify the `MIMode` are you setting it to `gdb` or `lldb` ?

**Comment by pieandcakes** [comment]
@kanflo So I don't have an arm device to test this with, but I verified that Linux to Linux or windows to Linux using `ssh` (pipeTransport) I'm able to both break (pause) with or without the stopAtEntry flag set.

**Comment by kanflo** [comment]
Thanks @pieandcakes. I would need to do some debugging before submitting and I am unsure if I have all the sources to build the entire cpptool extension. I am setting ```MIMode```to ```gdb```.

I must say I love VSCode for ARM debugging, I only need to have it break when needed without me resorting to OpenOCD scripting :)

**Comment by pieandcakes** [comment]
I don't think you need the entire extension. I need to add some instructions to our repo but you can build MIEngine and update the two files from the build and debug it. You will need to be setup to debug mono code as on non-Windows, that's the framework we use.

**Comment by kanflo** [comment]
If you find the time to add the build instructions I will find the time to dig into this issue. I have been longing for something else than TUI based GDB or bloated Eclipse and VSC seems to be it.

**Comment by pieandcakes** [comment]
@kanflo I'll see if I can get it added before Monday. Its been something I've wanted to do for a while and I think it might be a good time to try and prioritize it.

**Comment by pieandcakes** [comment]
@kanflo sorry I'm a bit late. Can you try the following doc and see if that works?

https://github.com/Microsoft/vscode-cpptools/blob/master/Documentation/Debugger/How%20To%20Debug%20MIEngine.md

**Comment by kanflo** [comment]
@pieandcakes, now worries! Thanks for the instructions, I will give them a try.
